code/ship/ship.cpp

```diff
@@ -15079,9 +15079,32 @@ SCP_string ship_return_orders(ship* sp)
 // of what a ship's orders are.  Feel free to use this function if 
 // it suits your needs for something.
 char *ship_return_time_to_goal(char *outbuf, ship *sp)
+{
+	int	minutes, seconds;
+	int time = ship_return_seconds_to_goal(sp);
+	if ( time >= 0 ) {
+		minutes = time/60;
+		seconds = time%60;
+		if ( minutes > 99 ) {
+			minutes = 99;
+			seconds = 99;
+		}
+		sprintf(outbuf, NOX("%02d:%02d"), minutes, seconds);
+	} else if ( time == -2) {
+        // -2 is used as a signal of "there is no valid goal right now"
```

Comment:
This is all well and good, but shouldn't something be done here?  What sort of contract do the callers expect?

Hint: I looked at where the function is used, and there are two use cases for which you return -2.  They are supposed to be handled differently.

Reply:
What do you recommend here?
The first -2 return case is when the computed distance is <1.0, presumably in meters.  So this may be assumed to be a short-circuit and should probably return a time estimate of 0s.

The second -2 return case is when the goal is not a waypoint or dock point, which could either return 0s or Unknown.